samples/azure-quantum/chemistry/QuantumPhaseEstimation/Molecule.ipynb

```diff
@@ -0,0 +1,1775 @@
+      "text/plain": [
+       "<Figure size 576x432 with 1 Axes>"
+      ]
+     },
+     "metadata": {
+      "needs_background": "light"
+     },
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "import matplotlib\n",
+    "import pylab as pl\n",
+    "%matplotlib inline\n",
+    "\n",
+    "pl.figure(figsize=(8, 6))\n",
+    "pl.bar(\n",
+    "    [\n",
+    "        \"Caffeine, QPE\",\n",
+    "        \"Caffeine, VQE\",\n",
+    "        \"H2, QPE x1000\",\n",
+    "        \"H2, VQE x1000\",\n",
+    "    ],\n",
+    "    [\n",
+    "        resources_caffeine_qpe[\"QubitClifford\"],\n",
+    "        resources_h2_qpe[\"QubitClifford\"],\n",
+    "        resources_caffeine_vqe[\"QubitClifford\"] * 1e3,\n",
+    "        resources_h2_vqe[\"QubitClifford\"] *  1e3,\n",
+    "    ]\n",
+    ")\n",
+    "pl.ylabel(\"Number of qubit Clifford gates\")\n",
+    "pl.title(\"Resources for H2 vs Caffeine\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Estimate ground state\n",
+    "\n",
+    "The smallest circuit of the 4 aforementioned is VQE on H2. However, it's still quite large and will require some state-of-the-art fine-tuned quantum hardware. To be able to run the sample on hardware, let's try to make it even smaller."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "prepare_state = qsharp.compile(\"\"\"open Microsoft.Quantum.Core;\n",
+    "open Microsoft.Quantum.Chemistry;\n",
+    "open Microsoft.Quantum.Chemistry.JordanWigner;\n",
+    "open Microsoft.Quantum.Chemistry.JordanWigner.VQE;\n",
+    "open Microsoft.Quantum.Intrinsic;\n",
+    "open Microsoft.Quantum.Diagnostics;\n",
+    "\n",
+    "operation PrepareStateVQE (JWEncodedData: JordanWignerEncodingData, theta1: Double, theta2: Double, theta3: Double) : Unit {\n",
```

Comment:
Out of curiosity, what motivated compiling this one inline, versus the other operations defined in the local workspace?

Reply:
This way the code is shown in the notebook.

Reply:
Makes sense; wasn't sure if there was another reason I was missing, so figured I'd ask. Thanks!